### PR TITLE
Replace GA search events with Monorail

### LINKED ISSUE #8696: Add Monorail search events to docs
Mirror the .dev site search even tracking so that we have data for future .dev learnings

```[tasklist]
### Tasks
- [ ] https://github.com/Shopify/polaris/issues/8703
- [ ] https://github.com/Shopify/polaris/issues/8704
- [ ] https://github.com/Shopify/polaris/issues/8705
```

### PR DESCRIPTION
**Pull request by yurm04**
### WHY are these changes introduced?

Fixes [#8696](https://github.com/Shopify/polaris/issues/8696)

Replacing GA search events with Monorail because .dev uses Monorail and we can compare results a little better.

⚠️ Blocked by https://github.com/Shopify/monorail/pull/15320 and will need to update and match Monorail event types when #15320 is merged

### How to 🎩
Check the console for Monorail `click` and `query` events

Screenshot 2023-03-17 at 5 50 12 PM

Screenshot 2023-03-17 at 5 49 29 PM

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

File: polaris.shopify.com/package.json
Comment (yurm04) [inline polaris.shopify.com/package.json:1]: i added the `monorail` and `uuid` dependencies. I think these other changes are just re-ordering packages that already exist in the website, not adding net new packages

File: polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx [hunk 1]
```diff
@@ -133,6 +164,7 @@ function GlobalSearch() {
         setSearchResults(results);
       });
 
+    captureSearchQuery(uuid, searchTerm, resultsInRenderedOrder);
     setCurrentResultIndex(0);
     scrollToTop();
   }, 400);
```
Comment (yurm04) [inline polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx:187]: the Monorail report will automatically take the last timestamped event so it shouldn't be a big deal on that end if we have a bunch of these firing. Is it a big deal for us on the frontend?

File: polaris.shopify.com/package.json [hunk 2]
```diff
@@ -36,17 +37,18 @@
...
     "remark-gfm": "^3.0.1",
-    "react-syntax-highlighter": "^15.5.0",
     "remark-unwrap-images": "^3.0.1",
-    "use-dark-mode": "^2.3.1"
+    "sharp": "^0.31.2",
+    "use-dark-mode": "^2.3.1",
+    "uuid": "^9.0.0"
   },
   "devDependencies": {
     "@types/gtag.js": "^0.0.10",
```
Comment (martenbjork) [inline polaris.shopify.com/package.json:53]: Can we remove @types/gtag.js?

File: polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx [hunk 3]
```diff
@@ -156,6 +205,10 @@ function GlobalSearch() {
   useEffect(() => {
     if (!isOpen) {
       setSearchTerm('');
+      setUuid(''); // reset the uuid if the modal is closed
+    } else {
+      // set uuid when modal is opened
```
Comment (martenbjork) [inline polaris.shopify.com/src/components/GlobalSearch/GlobalSearch.tsx]: Nit: The code is clear enough without the comments 😀

**Review by martenbjork — APPROVED** [review]

**Pull request closed without merging**